cache/src/main/java/io/envoyproxy/controlplane/cache/SimpleCache.java

```diff
@@ -31,16 +32,35 @@
 
   private final Consumer<String> callback;
   private final NodeGroup groups;
+  private final ExecutorService executorService;
 
   private final Object lock = new Object();
   private final Map<String, Snapshot> snapshots = new HashMap<>();
   private final Map<String, Map<Long, Watch>> watches = new HashMap<>();
 
   private long watchCount;
 
+  /**
+   * Constructs a simple cache.
+   *
+   * @param callback callback invoked when a group is seen for the first time
+   * @param groups maps an envoy host to a node group
+   */
   public SimpleCache(Consumer<String> callback, NodeGroup groups) {
+    this(callback, groups, ForkJoinPool.commonPool());
```

Comment:
Probably worth calling out that this is using the common pool in the javadoc. If this was used in highly concurrent setting, it might also end up being a point of unexpected contention.

Alternatively, you could just do away with this constructor and let the called BYO executor.

Reply:
I'd totally be down to get rid of this constructor, I was really trying to not break existing clients (although I've done that in one of the other PRs...). It feels kinda strange for the class to be opinionated about what the right `Executor` is

Reply:
Yeah, like I mentioned in #26, I don't think we care too much about backwards compatibility quite yet, but this does seem like a scenario where the common pool is a sane default. Other than needing the aforementioned javadoc comment mentioning that it defaults to the common pool, LGTM.